server/server.go

```diff
@@ -40,6 +40,8 @@ type Config struct {
 	HTTPListenPort   int
 	GRPCListenPort   int
 
+	DontRegisterInstrumentation bool
```

Comment:
I did it this way so the zero value preserved the current behaviour.  This isn't something I made a flag for as I don't expect people will need to set this on the command line, and therefore I have no way of setting a default value of true.

I could make a flag with a default of true if you think that would be better - WDYT?